### DISCUSSION TITLE
vite project showing 404 pages when uploaded to github pages

### DISCUSSION THREAD

**Discussion by saram7619** [community/community#61478] [Programming Help]
### Body

Hi Everyone,
I hope someone can help me as I have been trying to fix this problem for 3 days.
I have deployed a Vite project to github pages, but I get this error message 

File not found

The site configured at this address does not contain the requested file.

If this is your site, make sure that the filename case matches the URL as well as any file permissions.
For root URLs (like http://example.com/) you must provide an index.html file. 

Firefox dev tools is giving me this error
Content-Security-Policy: The page's settings blocked the loading of a resource at inline ("default-src"). 2 prepareInjection.js:249:78
Content-Security-Policy: The page's settings blocked the loading of a resource at https://saram7619.github.io/favicon.ico ("img-src"). 2 FaviconLoader.jsm:180:19

I have changed my config.js to base: "/deploy-vite/"

my repository is found
[repo](https://saram7619.github.io/deploy-vite/)

Does anyone know how to fix this?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by radar-blip** [comment]
It may be because you have everything inside a folder and nothing at root for Pages to find (such as index.html).

**Reply by saram7619** [reply]
My files are in my root folder, but still getting the 404 page and error.

**Reply by radar-blip** [reply]
But they are in a folder, you need to take them out of the folder (at least the index.html).

**Reply by mauriciohillesheim** [reply]
seu repositório está vazio https://github.com/search?q=saram7619&type=repositories

**Comment by karakuscem** [comment]
- Add : base: '/your-repo-name-here/', to vite.config.js,

- Run this command to add gh-pages as dependency `npm install gh-pages --save-dev`

- Add these two lines to package.json file under the scripts property
```
"predeploy": "npm run build",
"deploy": "gh-pages -d dist",
```
- And run `npm run deploy`
- Check your repository's pages settings, gh-pages branch should be selected.

**Reply by GhostOf0days** [reply]
1. index.html should be at the root for Vite projects.

2. Vite projects should use Netlify, Heroku, or Vercel.

**Reply by saram7619** [reply]
netilfy gives me the same blank page.

**Reply by AlexAlecky** [reply]
Am running "npm run deploy" but terminal is giving me an error because am running a wrong command

**Reply by tombohub** [reply]
instead of `base: '/your-repo-name-here/` just `base: ""` works for me.

**Reply by Rohit5055** [reply]
Added the following changes for me to work with gh-pages:

1. vite.config.js -> base: "/your-repo-name-here/",
2. App.tsx -> <BrowserRouter basename="/your-repo-name-here">
3. package.json -> Add a homepage property in this format*: https://{username}.github.io/{repo-name}
4. index.html ->

```
<script type="text/javascript">
      // Single Page Apps for GitHub Pages
      // https://github.com/rafgraph/spa-github-pages
      (function(l) {
        if (l.search[1] === '/' ) {
          var decoded = l.search.slice(1).split('&').map(function(s) { 
            return s.replace(/~and~/g, '&')
          }).join('?');
          window.history.replaceState(null, null,
              l.pathname.slice(0, -1) + decoded + l.hash
          );
        }
      }(window.location))
    </script>
```

5. Add a 404.html->
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>your-repo-name-here</title>
    <script type="text/javascript">
      // Single Page Apps for GitHub Pages
      // MIT License
      // https://github.com/rafgraph/spa-github-pages
      var pathSegmentsToKeep = 1;

      var l = window.location;
      l.replace(
        l.protocol + '//' + l.hostname + (l.port ? ':' + l.port : '') +
        l.pathname.split('/').slice(0, 1 + pathSegmentsToKeep).join('/') + '/?/' +
        l.pathname.slice(1).split('/').slice(pathSegmentsToKeep).join('/').replace(/&/g, '~and~') +
        (l.search ? '&' + l.search.slice(1).replace(/&/g, '~and~') : '') +
        l.hash
      );
    </script>
  </head>
  <body>
  </body>
</html>
```

**Comment by saram7619** [comment]
Thank you everyone for your suggestions. I have included my files.

deploy-vite/dist/index.html
`<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/deploy-vite/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React</title>
    <script type="module" crossorigin src="/deploy-vite/assets/index-18c8291d.js"></script>
    <link rel="stylesheet" href="/deploy-vite/assets/index-04490b11.css">
  </head>
  <body>
    <div id="root"></div>
    
  </body>
</html>
`
deploy-vite/index.html
`<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite + React</title>
  </head>
  <body>
    <div id="root"></div>
    <script type="module" src="/src/main.jsx"></script>
  </body>
</html>`

deploy-vite/package.json
`{
  "name": "deploy-vite",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "predeploy": "npm run build"
    "deploy": "gh-pages -d dist",
    "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview"
  },
  "dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@vitejs/plugin-react": "^4.0.3",
    "eslint": "^8.45.0",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "vite": "^4.4.5"
  }
}`

deploy-vite/vite.config.js
`import { defineConfig } from "vite"
import react from "@vitejs/plugin-react"

// https://vitejs.dev/config/
export default defineConfig({
  base: "/deploy-vite/",
  plugins: [react()],
})`

These are the files for the project that is giving me the 404 page.I have used netlify, that gives me the blank page. I have done the npm install gh-pages --save-dev, npm run deploy. I have started a new vite project with vite boilerplate and deployed to github pages. I have ran npm run preview which shows properly without errors. 
I just don't know how to fix this.

**Reply by GhostOf0days** [reply]
As many people have said, put your index.html at your project root not a folder.

**Reply by mtalha218** [reply]
Its in the root folder with src, public folder as it comes default. But still have this error when deployed on aws servers

**Reply by Ridlo543** [reply]
hello @saram7619 , 
I got the same error, 
but I was able to resolve it in my repository [my-repo](https://github.com/Ridlo543/IntelliTaskKanban.git)
the steps are as follows,
1. Previously the base in my config was `base: '/REPO/'` according to the documentation but got the error, then I changed vite.config.js to:
```javascript
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
  base: './',
  plugins: [react()],
})

```
However, you can follow the documentation to change the base to '/REPO/' or follow the base above

2. then, deploy according to the documentation instructions with github actions [vite.doc](https://vitejs.dev/guide/static-deploy.html) 
3. Or you can use the gh-pages package
- `npm install gh-pages --save-dev`
- add script package.json and `npm run deploy`
```
"deploy": "gh-pages -d dist",
```
- change the branch to (gh-pages) which will be deployed

**Reply by voidaugust** [reply]
Thank you, **base: './'** is the thing that worked for me.

**Reply by bonnerjallah** [reply]
i tried, base: './" and it still didn't work for me.

**Reply by rachpro1234** [reply]
Try to add a homepage key in your package.json file like this
"homepage": "http://<github-user-name>.github.io/<repo-name"

**Reply by rachpro1234** [reply]
thank you so much , the deploy key and the base was the reason why I couldn't deploy the app , your answer was so much helpful

**Reply by antra20** [reply]
thanks a lot ! it worked for me !!!

**Reply by CarroAlfred** [reply]
Thank you this worked for me

**Comment by sahilatahar** [comment]
You can also follow this repo [https://github.com/ErickKS/vite-deploy](https://github.com/ErickKS/vite-deploy) to deploy your react vite app

**Reply by anh3214** [reply]
I follow this but can't resole it?

**Comment by tombohub** [comment]
in vite config set `base: ""` to make assets relative path

**Comment by avanathan** [comment]
I am getting 404 `User attempted to access non-existent route` error while trying in enterprise github. In enterprise plan github pages are in `https://github.xyz.com/pages/<org>/<repo>`. I am using `./` as the `base`. Preview works fine. But when deploy to Pages it shows the error.

I wonder if there is a way to disable vite path check.

**Comment by avanathan** [comment]
@saram7619 Are you able to solve this problem?
